packages/core/src/new-api/internal/journal/memory-journal.ts

```diff
@@ -1,33 +1,26 @@
 import { JournalMessage } from "../new-execution/types/messages";
 
 import { Journal } from "./types";
-import { deserializeReplacer } from "./utils/deserialize-replacer";
 import { logJournalableMessage } from "./utils/log";
-import { serializeReplacer } from "./utils/serialize-replacer";
 
 /**
  * An in-memory journal.
  *
  * @beta
  */
 export class MemoryJournal implements Journal {
-  private messages: string[] = [];
+  private messages: JournalMessage[] = [];
 
   constructor(private _verbose: boolean = false) {}
 
   public record(message: JournalMessage): void {
     this._log(message);
 
-    this.messages.push(JSON.stringify(message, serializeReplacer.bind(this)));
+    this.messages.push(message);
```

Comment:
That was there to check that we didn't introduce any messages that we can't serialize.

Happy for it to go now, do you think it is worth while adding an equivalent test? Or are changes now likely to come with their own tests.